app/controllers/profiles_controller.rb

```diff
@@ -3,6 +3,9 @@ class ProfilesController < ApplicationController
   before_action :set_profile
 
   def update
+    weight = Profile.convert_imperial(params[:stones], params[:pounds])
+    # @profile.update(profile_params)
+    @profile.update(weight: weight)
```

Comment:
So note here that you're doing two saves to the database here - an `update` and a `save`, which would both generate update SQL statements. This also has the disadvantage that only weight can be updated.